irohad/multi_sig_transactions/state/impl/mst_state.cpp

```diff
@@ -107,6 +107,9 @@ namespace iroha {
    * @return return if at least one new signature was inserted
    */
   bool mergeSignaturesInBatch(DataType &target, const DataType &donor) {
+    if (target->transactions().size() != donor->transactions().size()) {
```

Comment:
Seems, the fix breaks functionality of MST. I guess it should be reworked with another approach, maybe remove `combine` or something else.